tests/python/unittest/test_gluon_estimator.py

```diff
@@ -275,3 +276,45 @@ def test_context():
                                   loss=loss,
                                   metrics=metrics,
                                   context='cpu')
+
+
+def test_batch_size():
+    '''Test batch size'''
+    num_samples = 32
+
+    # No Data Loader
+    data = mx.nd.random.uniform(shape=(num_samples, 3, 28, 28))
+    label = mx.nd.random.randint(low=0, high=2, shape=(num_samples,))
+    data_iter = mx.io.NDArrayIter(data=data, label=label, batch_size=16)
+    net = get_model()
+    loss = mx.gluon.loss.L2Loss()
+    ctx = mx.cpu()
+    est = estimator.Estimator(net=net, loss=loss, context=ctx)
+    with assert_raises(ValueError):
+        est.fit(train_data=data_iter)
+
+    # Empty data loader
+    data = mx.nd.random.uniform(shape=(0,))
+    label = mx.nd.random.randint(low=0, high=2, shape=(0,))
+    batch_size = 2
+    data_arr = mx.gluon.data.dataset.ArrayDataset(data, label)
+    data_loader = mx.gluon.data.DataLoader(data_arr, batch_size=batch_size)
+    est = estimator.Estimator(net=net, loss=loss, context=ctx)
+    with assert_raises(ValueError):
+        est.fit(train_data=data_loader)
+
+    # Batch size less than context
+    ctx = [mx.gpu(0), mx.gpu(1), mx.gpu(2), mx.gpu(3)]
```

Comment:
Are we sure we have 4 GPUs for unit tests? This might fail otherwise

Reply:
This specific test checks the total context should be less than the batch_size to equally split and loaded the data. It doesn't execute the forward/backward pass since the assert statement is at the start of fit() method. 

What could be the best way to execute this specific test case? Should it be only run on GPU machine by checking it as `mx.context.num_gpus()`?

Reply:
The net is initialized when creating the estimator object which happens before fit. The test will fail there if at least 4 GPUs are not available. 

I think checking the number of GPUs first acc your suggestion should fix it, or we can move the test to nightly where GPUs are available